sdk/nodejs/cloudwatch/eventRuleMixins.ts

```diff
@@ -107,6 +107,7 @@ export class EventRuleEventSubscription extends lambda.EventSubscription {
             rule: this.eventRule.name,
             arn: this.func.arn,
             targetId: name,
+            eventBusName: this.eventRule.eventBusName.apply(x => x!),
```

Comment:
is it possible that `this.eventRule.eventBusName` can be undefined? When constructing an `EventRule`, `eventBusName` is an [optional input](https://github.com/pulumi/pulumi-aws/blob/master/sdk/nodejs/cloudwatch/eventRule.ts#L234), the docs does say that it takes `default` event bus when omitted but I am not sure if that makes sure that `eventBusName` is always populated. 

In case `this.eventRule.eventBusName` can be undefined:
```js
eventBusName: this.eventRule.eventBusName == undefined 
  ? undefined
  : this.eventRule.eventBusName.apply(x => x!),
```
LGTM otherwise

Reply:
It is possible it is undefined, and if it is, this code will result in passing that undefined value through into the outer `eventBusName` property, which is correct.

Unfortunately, our TypeScript typing of optional inputs is "wrong" per https://github.com/pulumi/pulumi/issues/6175 (with a proposed fix in https://github.com/pulumi/pulumi/pull/6323 that has likely bitrotted).  This code I added here works around that "wrong"-ness, while still behaving correctly.

The code you suggested unfortunately won't work - because `this.eventRule.eventBusName` itself is not undefined, it is an `Output<undefined>`.  It is the value inside the Output which might be undefined.

Reply:
Aah I see, thanks for explaining the context behind this! 

> The code you suggested unfortunately won't work

Yeah I was assuming the outer output itself could be undefined but it seems that only the data underneath could be that.